src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/DefaultProcessRunner.java

```diff
@@ -87,8 +93,14 @@ public void run(String[] command) throws ProcessRunnerException {
 
       Process process = processBuilder.start();
 
-      handleStdOut(process);
-      handleErrOut(process);
+      // Only handle stdout or stderr if there are no listeners or if output wasn't redirected,
+      // which causes process.getInputStream()/process.getErrorStream() to return NullInputStream.
+      if (!(stdOutLineListeners.isEmpty() || inheritStdOutErr)) {
```

Comment:
In that situation, I prefer we log a message stating that the listeners won't receive anything. Do you prefer an actual exception?